### PR TITLE
Updates to Release documentation

### PR DESCRIPTION
**Pull request by lunaleaps**
This PR is about updating the existing release documentation to our current model

Notable updates are:
- Pick requests are now issues in this repo vs. comments on Discussions
- Release regression issues are now filed in this repo. See https://github.com/facebook/react-native/pull/43295 for changes in react-native/issues
- We are using Github Projects to triage Pick Requests / Issues. See https://github.com/reactwg/react-native-releases/projects?query=is%3Aopen

Todo:
- [x] Updated support policy -- updated these to new agreements internally and escalation.
- [x] Updated release roles -- updated to what we've discussed internally
- [x] Update general release process -- these instructions are basically the same for most releases. For release candidates, they'll be some differences, but I'm thinking release candidate (cut, patch, promote to stable) will reference a lot of these teps.
- [x] Consolidate release candidate instructions into one page and update

I'm going to merge this for now but plan to complete the following in a separate update:
- [ ] Update release testing docs
- [ ] Remove release documentation from reactnative.dev

### CODE REVIEW

File: docs/dependencies.md [hunk 1]
```diff
@@ -1,5 +1,7 @@
 # Release Dependencies
 
+
+
```
Comment (huntie) [inline docs/dependencies.md:4]: nit: Remove added newlines

File: docs/glossary.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
+
+$ Glossary
```
Comment (huntie) [inline docs/glossary.md:2]: Is this formatting (`$`) intentional?

**Review by huntie — APPROVED** [review]
LGTM, wow! ✅

Small nits/checking formatting. Please rebase + merge.

File: docs/faq.md [hunk 3]
```diff
@@ -49,4 +49,23 @@ We have a dedicated [release discussions repo](https://github.com/reactwg/react-
...
+
+At this point in time, we are prioritizing issues that are related to:
+
+* Latest version of React Native and two previous minor series.
+* Use of the New Architecture
+* Use of the Hermes Engine
+
+### Security Issues
+
+Meta has [a bounty program](https://www.facebook.com/whitehat/) for the safe disclosure of security bugs. In those cases, please go through the process outlined on that page and do not file a public issue.
```
Comment (cortinico) [inline docs/faq.md:71]: I think that those 2 are relevant to stay on the website as they're not specific to releases but related to our contribution process as a whole

**Pull request merged**